### PR TITLE
update: plugin API reference to version 3.6.0

### PR DESCRIPTION
**Pull request by mrgian**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind user-interface

/kind content

> /kind event

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area blog

/area documentation

> /area community

**What this PR does / why we need it**:
Updates the plugin API reference to version 3.6.0.
Updates docs about enabling plugin metrics in Falco.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Please note that this documents features available in Falco 0.38.1, so it should be merged after the Falco release.

### CODE REVIEW

**Comment by mrgian** [comment]
cc @FedeDP @incertum @jasondellaluce

File: content/en/docs/reference/plugins/plugin-api-reference.md
Comment (FedeDP) [inline content/en/docs/reference/plugins/plugin-api-reference.md:19]:
```suggestion
The C header files [plugin_api.h](https://github.com/falcosecurity/libs/blob/0.17.2/userspace/plugin/plugin_types.h) numerate all the API functions and associated structs/types as they are used by the plugins framework. The whole plugin API and the loader used in Falco are implemented in C in a standalone module located inside [falcosecurity/libs/userspace/plugin](https://github.com/falcosecurity/libs/tree/master/userspace/plugin), and can be imported and reused in other projects using the falcosecurity plugin system (e.g. we have a [plugin loader written in Go](https://github.com/falcosecurity/plugin-sdk-go/tree/main/pkg/loader) developed on top of the C one).
```

File: content/en/docs/metrics/falco-metrics.md
Comment (incertum) [inline content/en/docs/metrics/falco-metrics.md:1276]: Add below as paragraph:

Lastly, please note that as of Falco 0.38.1, it is not possible to fully use the remaining metrics feature when running plugins without the syscalls source. This is due to some regressions that prevent us from serving a small subset of metrics in this scenario. Near-term improvements in this regard are tracked in the following [issue]( https://github.com/falcosecurity/falco/issues/3194#issuecomment-2111009270). Please also keep in mind that many of the metrics categories are specific to the syscalls source, such as `libbpf_stats_enabled`, `kernel_event_counters_enabled`, or `state_counters_enabled`.

File: content/en/docs/reference/plugins/plugin-api-reference.md
Comment (incertum) [inline content/en/docs/reference/plugins/plugin-api-reference.md:237]:
```suggestion
	ss_plugin_metric_type type; // Metric type indicates whether the metric value is monotonic or non-monotonic.
```

File: content/en/docs/reference/plugins/plugin-api-reference.md
Comment (incertum) [inline content/en/docs/reference/plugins/plugin-api-reference.md:239]:
```suggestion
	ss_plugin_metric_value_type value_type; // Metric value data type, e.g. `uint64_t`
```

File: content/en/docs/reference/plugins/plugin-api-reference.md
Comment (incertum) [inline content/en/docs/reference/plugins/plugin-api-reference.md:790]:
```suggestion
Table accessors can be only obtained at initialization time, however tables may be empty at this time, which means subtables may be not yet exist.
```

Comment (incertum) [reply content/en/docs/reference/plugins/plugin-api-reference.md:790]: Could you provide clear examples how to use the new API instead of just providing the source code snippets?

For example how to access the fd table fields and the cmdline args and how is it different from accessing the currently available fields (see below)?

```
Header:
falcosecurity::table_field m_comm;

CPP:
init(falcosecurity::init_input& in)
{
    using st = falcosecurity::state_value_type;
    auto& t = in.tables();
    m_thread_table = t.get_table(THREAD_TABLE_NAME, st::SS_PLUGIN_ST_INT64);
    m_comm = m_thread_table.get_field(t.fields(), "comm", st::SS_PLUGIN_ST_STRING);
...

std::string tstr;
m_comm.read_value(tr, thread_entry, tstr);
```

Comment (mrgian) [reply content/en/docs/reference/plugins/plugin-api-reference.md:790]: There is already an example that shows how to obtain subtable accessors.
Once you obtained accessors and subtables handles, accessing subtables fields works just like any other regular table.
But I agree on making this more clear by providing another example 👍🏻

**Review by leogr — DISMISSED** [review]
LGTM.

/hold for a 2nd thought from previous reviewers

File: content/en/docs/reference/plugins/plugin-api-reference.md
Comment (incertum) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: It may seem intuitive to the folks who developed it. For end users who seek to develop a plugin the docs are not sufficient IMO. Instead of source code snippets from libs, what would be better would be clear examples on how to use the API similar to the snippet above.

Comment (leogr) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: I agree with @incertum . 

Just a question: do we want to address this point in this PR? Or do we want to unblock this PR and enhance its content in a follow-up PR? :thinking:

Comment (mrgian) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: The last commit already adds [an example](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#accessing-subtables) that shows how to access subtable fields. 
There is also [this example](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#obtaining-subtables-accessors) that shows how to obtain subtable accessors. 
These two examples are what the user needs to understand how to access subtables IMHO.

Please tell me if I am missing something 🤔

Comment (leogr) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: LGTM. 

@incertum ?

Comment (incertum) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: Hm the links seem to show libs snippet again? And does not appear to be an example of how to use it in a real plugin developed with the plugin SDK in the plugins repo.

Given the below example that shows how I read the `comm` value how do I read `m_args`, and `m_fd_name` values?

```
Header:
falcosecurity::table_field m_comm;

CPP:
init(falcosecurity::init_input& in)
{
    using st = falcosecurity::state_value_type;
    auto& t = in.tables();
    m_thread_table = t.get_table(THREAD_TABLE_NAME, st::SS_PLUGIN_ST_INT64);
    m_comm = m_thread_table.get_field(t.fields(), "comm", st::SS_PLUGIN_ST_STRING);
...

std::string tstr;
m_comm.read_value(tr, thread_entry, tstr);
```

Comment (mrgian) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: @incertum 
🤔 I think you are confusing plugin examples written using the plugin API, with libs snippets.
The purpose of the `Falco Plugins API Reference` page is to document symbols/types of the plugin API which is a plain-C interface.
For this reason, the example I provided shows how to use the plugin API directly without using the C++/Go plugin SDK.

You mentioned accessing the `fd_name` field. The example I provided shows exactly how to do that: [here](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#obtaining-subtables-accessors) it shows how to get the field accessor to that field, and [here](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#accessing-subtables) it shows how to use that accessor to obtain the actual `fd_name` value from one of the subtables.

Again, this example shows how to use the the plugin API and not the plugin SDK.
Documenting the plugin SDK is out of the scope of this website page IMHO. 
Also support for subtables in the C++ plugin SDK has not been merged yet (https://github.com/falcosecurity/plugin-sdk-cpp/pull/38) 

From what I know, the C++ SDK is still an experimental project, it can be subject to breaking changes and it has not been released. For this reason it has no documentation yet, however it will provide [examples](https://github.com/falcosecurity/plugin-sdk-cpp/pull/38/files#diff-c408a815df00653669c8ace735b01a6c91329e5537ab0bf32ca7dbbe8d17d15eR140).

Comment (incertum) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: Hey I am more interested in how to actually use it in a plugin, I added bunch of comments to my open PR https://github.com/falcosecurity/plugins/pull/419#discussion_r1649282173. Appreciate clear examples once the SDK is updated.

Comment (mrgian) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: I'll provide some examples when the C++ SDK is ready :)
But I think it should be addressed in another PR, since the goal of this one is updating the plugin API reference.

Comment (leogr) [reply content/en/docs/reference/plugins/plugin-api-reference.md]: Ok, I'd like to defer this to a follow-up PR and unblock this meantime.

**Review by leogr — APPROVED** [review]
/unhold

**Pull request merged**
